Src/PRuntimes/PSymRuntime/src/main/java/psym/runtime/machine/buffer/ReceiverQueue.java

```diff
@@ -0,0 +1,25 @@
+package psym.runtime.machine.buffer;
+
+import psym.runtime.machine.events.Message;
+import psym.valuesummary.Guard;
+
+import java.io.Serializable;
+
+/** Implements the Receiver Queue used to keep track of events with receiver queue semantics */
+public class ReceiverQueue extends SymbolicQueue implements Serializable {
```

Comment:
I dont understand this. Why are we adding receiver queue?

Reply:
I came across some examples where PSym with explicit search reports a non-trivial bug, while bugfinding doesn't in the same time limit. I want to confirm it is indeed a bug and not something broken in PSym. This is happening more often now and with long traces (> 50 steps).

I added printing a *.schedule file with PSym so that we can independently replay the trace with C# backend and confirm the trace with the much stable C# backend. This is going to be a solid feature as we give PSym to our users.

However, C#'s receiver queue semantics is not compatible with PSym's sender-queue semantics, making independently replaying a trace from PSym difficult. Since receiver-queue semantics is the default in P, I added it to PSym (rather easily as an option). Now, this enables generating traces from PSym that can be directly replayed with C# backend independently. Moreover, it will enable measuring how much savings we get from sender-queue semantics (I do not expect much for explicit search).

The alternative would have been translating a trace generated with sender-queue semantics into receiver-queue, which after some thoughts, doesn't look simple.

Now with PSym traces that can be independently replayed with C# backend, we gain more confidence on PSym implementation as well as can use all work on trace visualization for C# for free with no additional efforts.

Reply:
I do not think this is the right way of handing this. Lets discuss this offline.

Reply:
Removed receiver queue for now.